### DISCUSSION TITLE
검색 기능

### DISCUSSION THREAD

**Discussion by NamgyungKim** [prgrms-fe-devcourse/FEDC2_2JAVATAYO_Dongyoung#110] [Ideas]
![image](https://user-images.githubusercontent.com/87519250/173494908-272b25d9-2c6a-40ac-a125-b1a1a3204ae1.png)

1. 하단 태그 버튼
   - [x] 1 라디오버튼 (중복선택 불가능)
   - [ ] 2 체크박스버튼 (중복선택가능)

2. 처음 메인페이지들어오자마자 보여줄카드 
   - [ ] 1 전체를 채널을 보여주자 (단점. 모든 채널별로 for을 돌면서 post를 받아와야함)
   - [x] 2 하나를 선택해서 그것만 보여주자 (단점. 모든사람이 프론트 먼저 봐야함)
  (아쉬운점.. 처음에 로그인할때 사용자가 프론트인지 백인지 선택하는 부분이 있었으면 로그인했을때 바로 그 부분을 선택해놓은상태로 보여주면 좋았을텐데...)

3. 상단 텍스트 검색 시
   - [ ] 1 하단태그선택 한 상태로 필터를 해준다.
   - [x] 2 하단 태그선택을 다 취소시키고 전체 post제목으로 검색한다.

**Comment by NamgyungKim** [comment]
제 의견은 
1- 1
2- 2
3- 1 or 2
입이다~

**Comment by Hyevvy** [comment]
2번 아쉬운점에 대해서는 사용자가 원하는 직군을 여러개 선택했을 때 무엇을 보여줄 지 정하기 어렵다고 생각합니다.

**Comment by YuHyun-P** [comment]
✅ `3. 상단 텍스트 검색 시` 결정에 대한 추가 설명

**페이지 디자인 변경**
- 검색 바는 헤더로 옮기고, 전체 채널 검색으로 결정
  <img src="https://user-images.githubusercontent.com/96400112/173506261-48f861c3-122e-4092-a43a-15c72093cdfa.png" width="60%" />
- 검색 시 검색 결과 페이지로 이동
 
  ![Search](https://user-images.githubusercontent.com/96400112/173506517-050a6e68-61a5-404c-b015-0de35d55ec33.jpg)
